### PR TITLE
profile page

### PR DESCRIPTION
**Pull request by davidvargaspuga**
adds servlet for authenticating user and loading a profile. servlets return dummy data. checks if user is logged in: asks to log in for contact info. adds diego's styling for projects and header tabs and diego's parsing for projects.

### CODE REVIEW

**Comment by Manniec** [comment]
I went ahead and merged Diego's pull request to master 1st. When I went to approve this one there was a merge conflict bc Diego's code had some stuff in a the SearchServlet.java file that David's didn't. Not a big conflict. Bc I tried to resolve the merge error though someone else has to approve it before it can merge.  :)

File: src/main/java/com/google/sps/data/Profile.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+import java.util.*;
+
+/** An item on a todo list. */
```
Comment (marcelpuyat) [inline src/main/java/com/google/sps/data/Profile.java]: Update this comment

File: src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.*;
+import java.lang.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing tasks. */
```
Comment (marcelpuyat) [inline src/main/java/com/google/sps/servlets/AuthenticationServlet.java]: Update comment

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing tasks. */
+@WebServlet("/profile")
+public class ProfileInfoServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // initialize Profile with dummy data
```
Comment (marcelpuyat) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java:36]: It's really helpful to insert a TODO: <what should be done here to eventually use real data> as a reminder either for yourself when you get back to this, or as a note for someone else if someone ends up taking over this work (not likely for this project but this often happens in other environments).

File: src/main/webapp/profile.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            <tr>
+                <h2 id="name">Name</h2>
+            </tr>
+            <tr>
+                <p id="contact-info">Contact Info</p>
+            </tr>
+            <tr>
+                <!-- <td>
+                    <ul id="project-list">Project History</ul>
+
```
Comment (marcelpuyat) [inline src/main/webapp/profile.html]: Please leave context for commented out code – makes it easier for others to understand why this is commented out.

File: src/main/webapp/profile.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+/** Fetches the profile from the server and adds them to the DOM. */
+function loadProfile() {
+    fetch('/profile').then(response => response.json()).then((profile) => {
+
+        // initialize elements
+        const nameElement = document.getElementById("name");
+        const contactInfoElement = document.getElementById("contact-info");
+        const reviewListElement = document.getElementById("review-list");
+        // const projectListElement = document.getElementById("project-list");
```
Comment (marcelpuyat) [inline src/main/webapp/profile.js]: Same as above

File: src/main/webapp/profile.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        // display profile
+        nameElement.innerText = profile.name;
+        console.log(profile.name);
+        contactInfoElement.innerHTML = profile.contactInfo.Email + "<br/>" + profile.contactInfo.Mobile;
+        console.log(profile.contactInfo.Email + "<br/>" + profile.contactInfo.Mobile);
+        contactInfoElement.display = "none";
+
+        
+        /*
+        Below is ommited for now
```
Comment (marcelpuyat) [inline src/main/webapp/profile.js]: Same as above, even something as simple as "This code can be uncommented once we support fetching projects associated with a user".

In the real world you'd usually have a bug/feature request associated with each task, so you could add context by just referencing a link to said bug/feature request.

File: src/main/webapp/profile.js [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+        const loggingElement = document.getElementById("logging");
+
+        var isLoggedIn = (authentication.isLoggedIn == 'true'); 
+        var loginUrl = authentication.login;
+        var logoutUrl = authentication.logout;
+        if(isLoggedIn){
+            loggingElement.innerText = "LOG OUT";
+            loggingElement.href = loginUrl;
+            contactInfoElement.display = "inline";
+            // logoutElement.innerHTML = "<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>"
```
Comment (marcelpuyat) [inline src/main/webapp/profile.js]: Same as above

File: src/main/java/com/google/sps/data/Profile.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+public final class Profile {
+
+  private final String name;
+  private final ArrayList<String> projectHistory;
+  private final ArrayList<String> reviews;
+  private final HashMap<String, String> contactInfo;
```
Comment (DIEGOTEC18) [inline src/main/java/com/google/sps/data/Profile.java:24]: This is nice. Great use of HashMaps to save all contact info types (email, phone, social networks, etc.)

File: src/main/webapp/profile.js [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+        } else {
+            loggingElement.innerText = "LOG IN";
+            loggingElement.href = loginUrl;
+            contactInfoElement.innerHTML = "Login <a href=\"" + loginUrl + "\">here</a> to see the contact info.";
+            contactInfoElement.display = "inline";
+        }
+    });
+}
+
+/* Author: Diego V.A. */
```
Comment (DIEGOTEC18) [inline src/main/webapp/profile.js:70]: ;)

File: src/main/webapp/profile.js [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+    });
+
+    // call other loading 
+    loggingIn();
+    loadProjects();
+}
+
+/* displays contact info if logged in. otherwise, asks to log in*/
+function loggingIn(){
+    fetch('authentication').then(response => response.json()).then((authentication) =>{
```
Comment (DIEGOTEC18) [inline src/main/webapp/profile.js]: I think you missed a '/'. Shouldn't it be `fetch('/authentication')`?

File: src/main/webapp/profile.js [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+            loggingElement.innerText = "LOG IN";
+            loggingElement.href = loginUrl;
+            contactInfoElement.innerHTML = "Login <a href=\"" + loginUrl + "\">here</a> to see the contact info.";
+            contactInfoElement.display = "inline";
+        }
+    });
+}
+
+/* Author: Diego V.A. */
+function loadProjects(){
```
Comment (DIEGOTEC18) [inline src/main/webapp/profile.js:71]: I modified this function in my last pull request to separate the fetch part (now with filters) from the parsing (build project cards) part. Maybe you can repurpose it in the future to filter by project author only.

**Review by DIEGOTEC18 — APPROVED** [review]
Looks good to me! I don't know much about user auth with the gcloud API but it looks good!

**Review by marcelpuyat — APPROVED** [review]
ugh i guess i forgot to hit "submit review"... sorry about that

**Pull request merged**